app/scripts/utils/utils.ts

```diff
@@ -44,3 +48,31 @@ export function isEqualObj(a, b) {
   if (a === b) return true;
   return JSON.stringify(a) === JSON.stringify(b);
 }
+
+/**
+ * Adds a visuallyDisabled prop to the component that when set to true visually
+ * disables the component (through css) and prevents a click.
+ *
+ * @param Comp React Component to enhance
+ * @returns Enhanced styled component.
+ */
+export function composeVisuallyDisabled(Comp) {
+  return styled(Comp).attrs((props) => {
+    const onClickOriginal = props.onClick;
+    return {
+      onClick: (e) => {
+        if (props.visuallyDisabled) {
+          e.preventDefault();
+          return;
+        }
+        return onClickOriginal?.(e);
+      }
+    };
+  })`
+    ${({ visuallyDisabled: vd }) =>
+      vd &&
+      css`
+        ${visuallyDisabled()}
+      `}
+  `;
+}
```

Comment:
@nerik To show a tooltip over a disabled button, it cannot be disabled, otherwise the pointer events are removed and the tooltip is not triggered. Visually disabling the button has the same visual result, but the click event still fires. This function allows us to compose components, disabling the click when visually disabled.

Perhaps this should also add the `aria-disabled` role, even though the element can still be interacted with to show a tooltip. @ricardoduplos thoughts here?

Reply:
@danielfdsilva I had the same issue and used a [very mediocre hack](https://github.com/NASA-IMPACT/delta-ui/pull/284/commits/e269b4e67694873fe13e4118c2e15ccffd77e6e1#diff-111077f0776e65e2cf1a431485f118e091b99e82eb240ab5ad118c557de28323R18-R109).

Isn't there a simpler fix by just using a wrapper element? [Like this](https://codesandbox.io/s/withered-grass-yzhcry?file=/src/App.js)? Not sure it works with tippy

Reply:
I had not noticed that hack. 😅
Click events should only be used on interactive elements, like buttons, links, and the like. In this way the element is still focusable with keyboard navigation and can still be used, it just doesn't do anything. It is also semantically correct.

A couple of articles on the topic:
- [Semantic HTML: The Unbearable Rightness of Being](https://ux.shopify.com/semantic-html-the-unbearable-rightness-of-being-9b3c493e1791)
- [How (Not) to Build a Button](https://benmyers.dev/blog/clickable-divs/)

Full disclosure: I did not fully read the articles, but checked that they have the important parts

Reply:
regarding `aria` roles: Maybe tooltip can have `aria-live` role so it can notify users when it shows up?

Reply:
@danielfdsilva You are right. Will fix on an upcoming PR.